### PR TITLE
Update readme

### LINKED ISSUE #1814: [Documentation] Add README section on how to build / install pytorch and it's dependencies which will work with the Triton (top of the main branch or release branch)

### LINKED ISSUE #1815: [Documentation] Update README file - working driver versions, explicit notes on IPEX, oneAPI BaseKit 2024.x incompatibility

### PR DESCRIPTION
**Pull request by pbchekin**
Fixes #1814, #1815.

### CODE REVIEW

**Comment by pbchekin** [comment]
Rendered README.md: https://github.com/intel/intel-xpu-backend-for-triton/blob/47d1614622a67c40e4842cf2856875b6f5eea42d/README.md

File: README.md [hunk 1]
```diff
@@ -8,39 +8,96 @@ This is the development repository of Intel® XPU Backend for Triton\*, a new [T
...
+source /opt/intel/oneapi/setvars.sh
+```
 
 # Install from source
 
+## Prerequisites
+
+1. Latest [Rolling Release of GPU driver](https://dgpu-docs.intel.com/driver/installation.html)
+2. Latest release of [PyTorch Prerequisites for Intel GPUs](https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html)
+3. Latest release of [Profiling Tools Interfaces for Intel GPU (PTI for GPU)](https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html)
```
Comment (anmyachev) [inline README.md:57]: The prerequisites are the same, can there only be one copy of them?

Comment (pbchekin) [reply README.md:57]: Yes, they are the same now. I think it is simpler to have them in both sections (Quick install and Install from source) without jumping to the prerequisites in other place. Also note that while currently build and runtime prerequisites are the same in nearest future they can be different.

File: README.md [hunk 2]
```diff
@@ -8,39 +8,96 @@ This is the development repository of Intel® XPU Backend for Triton\*, a new [T
...
 
+```shell
+python -m venv .venv --prompt triton
+source .venv/bin/activate
+```
+
+Install build tools:
+
+```shell
+pip install ninja cmake wheel pybind11
```
Comment (anmyachev) [inline README.md:87]: This is not necessary unless we use `--no-build-isolation`. https://github.com/intel/intel-xpu-backend-for-triton/blob/3fc58dff56b558f3814beb038ba234aeb42e9ea1/python/pyproject.toml#L3 will handle build dependencies automatically.

Comment (pbchekin) [reply README.md]: Removed.

**Review by vlad-penkin — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -8,39 +8,90 @@ This is the development repository of Intel® XPU Backend for Triton\*, a new [T
...
+source /opt/intel/oneapi/setvars.sh
+```
 
 # Install from source
 
+## Prerequisites
+
+1. Latest [Rolling Release](https://dgpu-docs.intel.com/driver/installation-rolling.html) or [Long Term Support Release](https://dgpu-docs.intel.com/driver/installation.html) of GPU driver
+2. Latest release of [PyTorch Prerequisites for Intel GPUs](https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html)
+3. Latest release of [Profiling Tools Interfaces for Intel GPU (PTI for GPU)](https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html)
```
[the remark this answers is not in the document]
Comment (pbchekin) [reply README.md:57]: I do not have full context on this, sounds very confusing why a regular Triton user should care about PTI/Kineto. You can add this clarification in a separate PR.

File: README.md [hunk 4]
```diff
@@ -8,39 +8,90 @@ This is the development repository of Intel® XPU Backend for Triton\*, a new [T
...
-cd intel-xpu-backend-for-triton;
 
-python -m venv .venv --prompt triton;
-source .venv/bin/activate;
+Compile and install PyTorch:
 
-pip install ninja cmake wheel pybind11; # build-time dependencies
-pip install -e python
+```shell
+scripts/install-pytorch.sh --source
```
[the remark this answers is not in the document]
Comment (pbchekin) [reply README.md:87]: The preference is to use nightly wheels for both PyTorch and Triton. If a Triton user (not a developer) user goes with compiling from source, it is most likely to use Triton with different OS (RHEL, Rocky Linux, and so on). In this case, installing PyTorch from nightly wheels won't work. For a small number of developers on Ubuntu 22.04, who install PyTorch from nightly wheels but compile Triton from sources, we can add this option in the next iteration.

File: README.md [hunk 5]
```diff
@@ -8,39 +8,90 @@ This is the development repository of Intel® XPU Backend for Triton\*, a new [T
...
+1. Latest [Rolling Release](https://dgpu-docs.intel.com/driver/installation-rolling.html) or [Long Term Support Release](https://dgpu-docs.intel.com/driver/installation.html) of GPU driver
+2. Latest release of [PyTorch Prerequisites for Intel GPUs](https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html)
+3. Latest release of [Profiling Tools Interfaces for Intel GPU (PTI for GPU)](https://www.intel.com/content/www/us/en/developer/articles/tool/pytorch-prerequisites-for-intel-gpus.html)
+
+## Compile PyTorch and Triton from source
+
+Currently, Intel® XPU Backend for Triton\* requires a special version of PyTorch and both need to be compiled at the same time.
+
+Before compiling PyTorch and Intel® XPU Backend for Triton\* you need to initialize the toolchain.
+By default, it is installed to `/opt/intel/oneapi`:
```
[the remark this answers is not in the document]
Comment (pbchekin) [reply README.md]: Added a clarification on the default location if installed as root vs. user.

**Pull request merged**